libvast/src/query_queue.cpp

```diff
@@ -235,4 +235,15 @@ query_queue::handle_completion(const uuid& qid) {
   return result;
 }
 
+size_t query_queue::memusage() const {
+  auto usage = size_t{0u};
+  for (const auto& [uid, query_state] : queries_) {
+    usage += sizeof(uid) + sizeof(query_state);
```

Comment:
The `query_state` contains a `query_context` which in turn has multiple heap-stored members, I'm afraid we need a proper `memusage()` function for those.

Reply:
Do we wan't a memusage for expression? It seems that i must either visit it recursively or i need to have some counter that is updated and persisted on creation.

Reply:
I agree, it isn't worth the effort.